bbq/vm/vm.go

```diff
@@ -627,25 +625,42 @@ func opGetIndex(vm *VM) {
 func opInvoke(vm *VM, ins opcode.InstructionInvoke) {
 	typeArguments := loadTypeArguments(vm, ins.TypeArgs)
 
-	value := vm.pop()
+	functionValue := vm.pop()
+
+	explicitArgumentsCount := int(ins.ArgCount)
+	arguments := vm.peekN(explicitArgumentsCount)
+
+	// If the function is a pointer to an object-method, then the receiver is implicitly captured.
+	if boundFunction, isBoundFUnction := functionValue.(*BoundFunctionPointerValue); isBoundFUnction {
+		functionValue = boundFunction.Method
+		receiver := unwrapReceiver(vm.context, boundFunction.Receiver)
+		arguments = append([]Value{receiver}, arguments...)
+	}
```

Comment:
Now the instruction is responsible for passing the arguments to the `invokeFunction` method. See the below comment for details.